File: features/person/src/main/java/com/nivasafinance/features/person/exception/PersonNotFoundException.java
package com.nivasafinance.features.person.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import com.nivasafinance.common.exception.ResourceNotFoundException;
import org.springframework.context.MessageSource;

import java.io.Serial;

public class PersonNotFoundException extends ResourceNotFoundException {

    @Serial
    private static final long serialVersionUID = 932223434L;
    
    public PersonNotFoundException(Long personId, MessageSource messageSource) {
        super(ExceptionUtils.createLocalizedMessage(
                "error.person.id.not.found",
                new Object[]{personId.toString()},
                messageSource
        ));
    }
}

File: features/person/src/main/java/com/nivasafinance/features/person/exception/PersonNotFoundException.kt
package com.nivasafinance.features.person.exception

import com.nivasafinance.common.exception.ExceptionUtils
import com.nivasafinance.common.exception.ResourceNotFoundException
import org.springframework.context.MessageSource
import java.util.UUID

class PersonNotFoundException(
    personId: UUID,
    messageSource: MessageSource
) : ResourceNotFoundException(
    ExceptionUtils.createLocalizedMessage("error.person.not.found", arrayOf(personId.toString()), messageSource)
)
